nexus_constructor/InstrumentView.py

```diff
@@ -0,0 +1,157 @@
+
+        self.red_material.setAmbient(red)
+        self.red_material.setDiffuse(dark_red)
+
+        self.beam_material.setAmbient(blue)
+        self.beam_material.setDiffuse(light_blue)
+        self.beam_material.setAlpha(0.5)
+
+        self.green_material.setAmbient(grey)
+        self.green_material.setDiffuse(grey)
+
+    def create_sample_cube(self):
+        """
+        Creates the initial sample cube.
+        """
+        self.cube_mesh.setXExtent(1)
+        self.cube_mesh.setYExtent(1)
+        self.cube_mesh.setZExtent(1)
+
+        self.cube_entity.addComponent(self.cube_mesh)
+        self.cube_entity.addComponent(self.red_material)
+
+    def create_beam_cylinder(self):
+        """
+        Creates the initial beam cylinder.
+        """
+        self.cylinder_mesh.setRadius(2.5)
+        self.cylinder_mesh.setLength(self.cylinder_length)
+        self.cylinder_mesh.setRings(2)
+
+        cylinder_matrix = QMatrix4x4()
+        cylinder_matrix.rotate(270, QVector3D(1, 0, 0))
+        cylinder_matrix.translate(QVector3D(0, 20, 0))
+
+        self.cylinder_transform.setMatrix(cylinder_matrix)
+
+        self.cylinder_entity.addComponent(self.cylinder_mesh)
+        self.cylinder_entity.addComponent(self.beam_material)
+        self.cylinder_entity.addComponent(self.cylinder_transform)
+
+    def create_neutrons(self):
+        """
+        Creates the neutron animations.
+        """
+        # Create lists of x, y, and time offsets for the neutron animations
+        x_offsets = [0, 0, 0, 2, -2, 1.4, 1.4, -1.4, -1.4]
+        y_offsets = [0, 2, -2, 0, 0, 1.4, -1.4, 1.4, -1.4]
+        time_span_offsets = [0, -5, -7, 5, 7, 19, -19, 23, -23]
+
+        for i in range(self.num_neutrons):
+
+            # Create the neutron mesh and entity
+            neutron_entity = Qt3DCore.QEntity(self.rootEntity)
+            neutron_mesh = Qt3DExtras.QSphereMesh()
+            neutron_mesh.setRadius(3)
+
+            neutron_transform = Qt3DCore.QTransform()
+            neutron_animation_controller = NeutronAnimationController(
+                x_offsets[i], y_offsets[i], neutron_transform
+            )
+            neutron_animation_controller.set_target(neutron_transform)
+
+            # Instruct the NeutronAnimationController to move the neutron along the z-axis from -40 to 0
+            neutron_animation = QPropertyAnimation(neutron_transform)
+            neutron_animation.setTargetObject(neutron_animation_controller)
+            neutron_animation.setPropertyName(b"distance")
+            neutron_animation.setStartValue(-self.cylinder_length)
+            neutron_animation.setEndValue(0)
+            neutron_animation.setDuration(500 + time_span_offsets[i])
+            neutron_animation.setLoopCount(-1)
+            neutron_animation.start()
+
+            self.neutron_entities.append(neutron_entity)
+            self.neutron_meshes.append(neutron_mesh)
+            self.neutron_transforms.append(neutron_transform)
```

Comment:
Add a comment above these to mention that this is actually a workaround because there is a bug in the PySide2 bindings that requires us to keep these Qt3D entities in scope so they are not destroyed by C++. 

I wonder here if a dict might be better, as we are just using it to keep these variables in scope. It could be laid out like this for example: 
```python
entity_store = {
"neutron_entities" : [],
"neutron_meshes" : []
}
``` 
etc. 

this way when they fix this bug we can just remove the one dictionary and it's calls.